app/controllers/applications_controller.rb

```diff
@@ -1,6 +1,30 @@
 class ApplicationsController < ApplicationController
+
+  def index
+    @applications = Application.all
+  end
+  
+  def new
+    @application = Application.new
+  end
+
+  def create
+    @application = Application.new(application_params)
+    if @application.save
+      redirect_to "/applications/#{@application.id}"   
+    else
+      render :new
```

Comment:
I like the use of the render :new syntax to give the user another chance to input correct data.